server/sonar-server/src/main/java/org/sonar/server/component/index/ComponentIndex.java

```diff
@@ -107,27 +126,27 @@ private QueryBuilder createQuery(ComponentIndexQuery query, ComponentTextSearchF
       .setQueryText(query.getQuery())
       .setFieldKey(FIELD_KEY)
       .setFieldName(FIELD_NAME)
+      .setRecentlyBrowsed(query.getRecentlyBrowsed())
+      .setFavorites(query.getFavorites())
       .build();
     return esQuery.must(ComponentTextSearchQueryFactory.createQuery(componentTextSearchQuery, features));
   }
 
-  private static List<ComponentsPerQualifier> aggregationsToQualifiers(SearchResponse response) {
+  private static ComponentIndexResults aggregationsToQualifiers(SearchResponse response) {
     InternalFilters filtersAgg = response.getAggregations().get(FILTERS_AGGREGATION_NAME);
     List<Bucket> buckets = filtersAgg.getBuckets();
-    return buckets.stream()
-      .map(ComponentIndex::bucketToQualifier)
-      .collect(MoreCollectors.toList(buckets.size()));
+    return ComponentIndexResults.newBuilder()
+      .setQualifiers(
+        buckets.stream().map(bucket -> bucketToQualifier(bucket)))
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1612)